### PR TITLE
Fix query rows error bug

### PR DESCRIPTION
**Pull request by Seaven**
Backgroud:

```
DROP TABLE IF EXISTS test;
CREATE TABLE test (
  id int(11),
  name varchar(128)
) ENGINE=OLAP 
DUPLICATE KEY(id)
DISTRIBUTED BY HASH(id) BUCKETS 10 
PROPERTIES (
  "replication_num" = "1",
  "in_memory" = "true",
  "storage_format" = "DEFAULT"
);

insert into test
select 1,'a1' union all
select 2,'a2' union all
select 3,'a3' union all
select 4,'a4' union all
select 5,'a5' union all
select 6,'a6' union all
select 7,'a7' union all
select 8,'a8' union all
select 9,'a9';
```

Some users has report bugs when using DBeaver for SQL:
```
mysql>  select v1.* from ( select id from test order by id limit 5 ) v1 left join test v2 on v1.id = v2.id;
```

In DBeaver:
```
mysql>  select v1.* from ( select id from test order by id limit 5 ) v1 left join test v2 on v1.id = v2.id;
+------+
| id   |
+------+
|    2 |
|    1 |
|    3 |
|    4 |
+------+
4 rows in set (0.03 sec)
```

But in Mysql client:
```
mysql>   select v1.* from ( select id from test order by id limit 5 ) v1 left join test v2 on v1.id = v2.id;
+------+
| id   |
+------+
|    2 |
|    5 |
|    1 |
|    3 |
|    4 |
+------+
5 rows in set (0.05 sec)
```

The result will lose one rows, the reason about:
* DBeaver will send `set SQL_SELECT_LIMIT =  xx` first before send query, but Mysql client not, so the SQL which starrocks received will be more limit clause compared to MySQL Client.
* CBO order by clause merge multi limit clause has same bug

### CODE REVIEW

**Comment by Seaven** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/MergeLimitWithSortRule.java [hunk 1]
```diff
@@ -30,8 +30,9 @@ public boolean check(final OptExpression input, OptimizerContext context) {
         LogicalLimitOperator limit = (LogicalLimitOperator) input.getOp();
         LogicalTopNOperator sort = (LogicalTopNOperator) input.getInputs().get(0).getOp();
 
-        OptExpression result = new OptExpression(
-                new LogicalTopNOperator(sort.getOrderByElements(), limit.getLimit(), limit.getOffset()));
+        long l = limit.hasLimit() ? limit.getLimit() : -1;
+        long of = limit.hasOffset() ? limit.getOffset() : 0;
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/MergeLimitWithSortRule.java:34]: What't meaning of these two line codes?  limit default value is -1, offset default value is 0.

Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/MergeLimitWithSortRule.java:34]: limit default offset is -1, TOPN handle -1 offset is error

**Review by Youngwb — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/MergeLimitWithSortRule.java (hunk 1 above)
Comment (kangkaisen) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/MergeLimitWithSortRule.java:34]: So, what's the meaning of `long l = limit.hasLimit() ? limit.getLimit() : -1;`?

The default offset for limit why not 0?

Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/MergeLimitWithSortRule.java:34]: Limit 0 is really can be appear in sql, like select * from xx limit 0, it's right and result return 0 rows

Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/MergeLimitWithSortRule.java:34]: BTW, I think we can change offset default 0, and support static `NONE_LIMIT` and `NONE_OFFSET`

Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/MergeLimitWithSortRule.java:34]: I will modify it in next PR

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**